### ISSUE TITLE
Bluetooth: GATT find info response

### ISSUE THREAD

**Issue by kapi-no** [zephyrproject-rtos/zephyr#10105]
There is probably a bug in ```gatt_find_info_rsp``` function located in the ```subsys/bluetooth/host/gatt.c``` file. Similar GATT response functions may also be affected.

It is assumed in this function that the group of discovered descriptors uses the same type of UUID: either 16-bit or 128-bit one. Now, imagine following scenario:

There is 128-bit charactersteristic with 2 descriptors within it: first one with characteristic value and second one with CCC. In this case, it would be assumed that CCC is 128-bit uuid and would be wrongly parsed (resulting in the callback with ```NULL``` GATT attribute).

**Comment by hakonfam** [comment]
_Per response_ it is only valid to have either 16 bit or 128 bit. As specified in the Core spec 5.0. 

![firp](https://user-images.githubusercontent.com/1547348/45803971-2de9bf80-bcbb-11e8-90fd-c05404ea06b0.PNG)

**Comment by Vudentz** [comment]
Nope, it actually checks if attribute is of the same type an otherwise it stops:

https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/host/att.c#L437

I know it works since BlueZ does use it and I got correct responses.

**Comment by kapi-no** [comment]
Yes, you are both right. I noticed a problem with GATT discovery in the aforementioned scenario, but I incorrectly identified the cause of it. 

Actually, it seems that ```gatt_discover_next``` function (called after splitting the discovery in 2 steps: one for 16-bit descriptor and one for 128-bit descriptor) incorrectly adjusts handle range and doesn't permit to start the discovery procedure with the same start and end handle.

This is addressed in #10119
